File: backend/app/models/approval.py
from __future__ import annotations

from typing import Literal

from pydantic import BaseModel


class ApprovalDecisionRequest(BaseModel):
    approver: str
    reason: str | None = None


class ApprovalRecord(BaseModel):
    approval_id: str
    event_id: str
    proposed_action: dict
    status: Literal["pending", "approved", "rejected"]
    approver: str | None = None
    reason: str | None = None

File: frontend/src/types/approval.ts
// Mirrors backend/app/models/approval.py

export type ApprovalStatus = "pending" | "approved" | "rejected";

export interface ApprovalRecord {
  approval_id: string;
  event_id: string;
  proposed_action: Record<string, unknown>;
  status: ApprovalStatus;
  approver?: string | null;
  reason?: string | null;
}
